spec.bs

```diff
@@ -8375,66 +8378,74 @@ To <dfn>build trusted key value bidding signals request body</dfn> given a [=set
   1. Let |compressionGroupId| be |compressionIdMap|[|joiningOrigin|].
   1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
-    1. Let |compressionGroupMap|[|compressionGroupId|] be an empty [=map=], whose [=map/keys=] are integers
+    1.[=map/Set=] |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers
       and [=map/values=] are [=maps=].
-  1. Set |executionMode| to |group|'s [=interest group/execution mode=].
-  1. If |executionMode| equal to "`group-by-origin`", set |partitionId| to 0.
+  1. If |group|'s [=interest group/execution mode=] is "`group-by-origin`", set |partitionId| to 0.
   1. Otherwise:
     1. If |compressionGroupMap|[|compressionGroupId|] [=map/contains=] 0, set |partitionId| to
       [=map/size=] of |compressionGroupMap|[|compressionGroupId|].
     1. Otherwise, set |partitionId| to sum [=list/size=] of |compressionGroupMap|[|joiningOrigin|]
       and 1.
-  1. Set |interestGroupIdMap|[|group|'s [=interest group/name=]] to [=tuple=] of |compressionGroupId|
-    and |partitionId|.
-  1. If |compressionGroupMap|[|compressionGroupId|] does not [=map/contain=] |partitionId|:
-    1. Let |partition| be an empty [=partition=].
-    1. Set |partition|'s [=partition/id=] to |partitionId|.
-    1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=list=]
-      of [=strings=].
-    1. Set |namespace|["interest_group_names"] to [|group|'s [=interest group/name=]].
-    1. Set |namespace|["bidding_keys"] to |group|'s [=interest group/trusted bidding signals keys=].
-    1. Set |partition|'s [=partition/namespace=] to |namespace|.
-    1. Let |metadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
-    1. Set |metadata|["experiment_group_id"] to |experimentGroupId|.
-    1. Set |metadata|[|slotSizeParams|[0]] to |slotSizeParams|[1].
-    1. Set |partition|'s [=partition/metadata=] to |metadata|.
-    1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|] to |partition|.
+  1. [=map/Set=] |interestGroupIdMap|[|group|'s [=interest group/name=]] to [=tuple=] of
+    (|compressionGroupId|, |partitionId|).
+  1. If |compressionGroupMap|[|compressionGroupId|] does not [=map/contain=] |partitionId|, Set
+    |compressionGroupMap|[|compressionGroupId|][|partitionId|] with the following [=struct/items=]:
+    : [=bidding partition=]
+    :: a [=struct=] with the following [=struct/items=]:
+      : [=bidding partition/id=]
+      :: an integer records the identifier of this [=bidding partition=].
+
+      : [=bidding partition/namespace=]
+      :: a [=map=] whose [=map/keys=] are [=strings=] and [=map/values=] are [=list=] of [=strings=].
+
+      : [=bidding partition/metadata=]
+      :: a [=map=] whose [=map/keys=] and [=map/values=] are [=strings=].
+
+    1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|]'s [=bidding partition/namespace=]["interest_group_names"]

```

Comment:
no longer need these "set" steps, but instead put their values to the "::" rows of each field. That's the whole reason why we wanted that format for readability.